lib/buffer.js

```diff
@@ -351,7 +364,7 @@ Buffer.concat = function(list, length) {
     length = length >>> 0;
   }
 
-  var buffer = Buffer.allocUnsafe(length);
+  var buffer = allocate(length);
```

Comment:
@RReverser Hmm, please take a look at #7051/#7079… doing `s/Buffer.allocUnsafe/allocate/` would mean that the extra check for negative inputs would be removed.